pkg/prm/prm.go

```diff
@@ -52,3 +75,153 @@ func (*Prm) cacheTool(tool *Tool) error {
+		tmpls = p.FilterFiles(tmpls, func(f ToolConfig) bool { return f.Plugin.Id == toolName })
+	}
+
+	tmpls = p.filterNewestVersions(tmpls)
+
+	return tmpls
+}
+
+func (p *Prm) filterNewestVersions(tt []ToolConfig) (ret []ToolConfig) {
+	for _, t := range tt {
+		id := t.Plugin.Id
+		author := t.Plugin.Author
+		// Look for tools with the same author and id
+		tools := p.FilterFiles(tt, func(f ToolConfig) bool { return f.Plugin.Id == id && f.Plugin.Author == author })
+		if len(tools) > 1 {
+			// If the author/id template has 2+ entries, that's multiple versions
+			// check first to see if the return list already has an entry for this template
+			if len(p.FilterFiles(ret, func(f ToolConfig) bool { return f.Plugin.Id == id && f.Plugin.Author == author })) == 0 {
+				// turn the version strings into version objects for sorting and comparison
+				versionsRaw := []string{}
+				for _, t := range tools {
+					versionsRaw = append(versionsRaw, t.Plugin.Version)
+				}
+				versions := make([]*version.Version, len(versionsRaw))
+				for i, raw := range versionsRaw {
+					v, _ := version.NewVersion(raw)
+					versions[i] = v
+				}
+				sort.Sort(version.Collection(versions))
+				// select the latest version
+				highestVersion := versions[len(versions)-1]
+				highestVersionTemplate := p.FilterFiles(tools, func(f ToolConfig) bool {
+					actualVersion, _ := version.NewVersion(f.Plugin.Version)
+					return actualVersion.Equal(highestVersion)
+				})
+				ret = append(ret, highestVersionTemplate[0])
+			}
+		} else {
+			// If the author/id template only has 1 entry, it's already the latest version
+			ret = append(ret, t)
+		}
+	}
+
+	return ret
+}
+
+func (p *Prm) FilterFiles(ss []ToolConfig, test func(ToolConfig) bool) (ret []ToolConfig) {
+	for _, s := range ss {
+		if test(s) {
+			ret = append(ret, s)
+		}
+	}
+	return
+}
+
+// FormatTools formats one or more templates to display on the console in
+// table format or json format.
+func (*Prm) FormatTools(tools []ToolConfig, jsonOutput string) (string, error) {
+	output := ""
+	switch jsonOutput {
+	case "table":
+		count := len(tools)
+		if count < 1 {
+			log.Warn().Msgf("Could not locate any tools at %+v", viper.GetString("toolpath"))
+		} else if count == 1 {
+			stringBuilder := &strings.Builder{}
+			stringBuilder.WriteString(fmt.Sprintf("DisplayName:     %v\n", tools[0].Plugin.Display))
+			stringBuilder.WriteString(fmt.Sprintf("Author:          %v\n", tools[0].Plugin.Author))
+			stringBuilder.WriteString(fmt.Sprintf("Name:            %v\n", tools[0].Plugin.Id))
+			stringBuilder.WriteString(fmt.Sprintf("ProjectURL:      %v\n", tools[0].Plugin.UpstreamProjUrl))
+			stringBuilder.WriteString(fmt.Sprintf("Version:         %v\n", tools[0].Plugin.Version))
+			output = stringBuilder.String()
+		} else {
+			stringBuilder := &strings.Builder{}
+			table := tablewriter.NewWriter(stringBuilder)
+			table.SetHeader([]string{"DisplayName", "Author", "Name", "ProjectURL", "Version"})
```

Comment:
Looked a little bit odd in the terminal, all caps. Just an opinion though, so dismiss if you want

```suggestion
			table.SetHeader([]string{"DisplayName", "Author", "Name", "Project_URL", "Version"})
```